qm-selinux.spec.rpkg

```diff
@@ -0,0 +1,99 @@
+# For automatic rebuilds in COPR
+
+# The following tag is to get correct syntax highlighting for this file in vim text editor
+# vim: syntax=spec
+
+%global debug_package %{nil}
+
+# container-selinux stuff (prefix with ds_ for version/release etc.)
+# Some bits borrowed from the openstack-selinux package
+%global selinuxtype targeted
+%global moduletype services
+%global modulenames qm
+
+# Usage: _format var format
+# Expand 'modulenames' into various formats as needed
+# Format must contain '$x' somewhere to do anything useful
+%global _format() export %1=""; for x in %{modulenames}; do %1+=%2; %%1+=" "; done;
+
+Name: {{{ git_dir_name }}}
+Epoch: 101
+Version: {{{ git_dir_version }}}
+Release: 1%{?dist}
+License: GPLv2
+URL: https://github.com/containers/container-selinux
+Summary: SELinux policies for container runtimes
+VCS: {{{ git_dir_vcs }}}
+Source: {{{ git_dir_pack }}}
+BuildArch: noarch
+BuildRequires: container-selinux
+BuildRequires: make
+BuildRequires: git-core
+BuildRequires: pkgconfig(systemd)
+BuildRequires: selinux-policy >= %_selinux_policy_version
+BuildRequires: selinux-policy-devel >= %_selinux_policy_version
+Requires: selinux-policy >= %_selinux_policy_version
+Requires(post): selinux-policy-base >= %_selinux_policy_version
+Requires(post): selinux-policy-targeted >= %_selinux_policy_version
+Requires(post): policycoreutils
+Requires(post): libselinux-utils
+
+%description
+SELinux policy modules for use with container runtimes.
+
+%prep
+{{{ git_dir_setup_macro }}}
+sed -i 's/install: man/install:/' Makefile
+
+# Remove unavailable tokens
+%if 0%{?fedora} <= 37 || 0%{?rhel} <= 9
+sed -i '/user_namespace/d' qm.if
+%endif
+
+# selinux-policy on RHEL 8 isn't new enough
+%if !0%{?fedora} && 0%{?rhel} <= 8
+sed -i '/miscfiles_watch_localization_files/d' qm.if
+sed -i '/dev_remount_sysfs_fs/d' qm.if
+sed -i '/fs_all_mount_fs_perms_tmpfs/d' qm.if
+sed -i '/fs_all_mount_fs_perms_xattr_fs/d' qm.if
+sed -i '/fs_search_tracefs_dirs/d' qm.if
+sed -i '/watch watch_reads/d' qm.if
+sed -i 's/bpf perfmon//' qm.if
+%endif
+
+%build
+%{__make}
+
+%install
+# install policy modules
+%_format MODULES $x.pp.bz2
+%{__make} DESTDIR=%{buildroot} DATADIR=%{_datadir} install install-setup
+
+%pre
+
+%post
+
+%postun
+if [ $1 -eq 0 ]; then
+   %selinux_modules_uninstall -s %{selinuxtype} %{modulenames}
+fi
+
+%posttrans
+
+#define license tag if not already defined
+%{!?_licensedir:%global license %doc}
+
+%files
+%doc README.md
+%dir %{_datadir}/selinux
+%{_datadir}/selinux/*
+%{_mandir}/man8/*
+%dir %{_sharedstatedir}/qm
+%dir %{_sharedstatedir}/qm/rootfs%{_datadir}
+%dir %{_sharedstatedir}/qm/rootfs%{_datadir}/containers
+%{_sharedstatedir}/qm/rootfs%{_datadir}/containers/container_contexts
+%dir %{_libexecdir}/%{name}
+%{_libexecdir}/%{name}/install.sh
```

Comment:
@rhatdan added the install.sh script to `/usr/libexec/qm-selinux` . Please verify the change and remind me what else I need to add.

Reply:
> I think we should name this /usr/libexec/qm-install 

I think the script should be 1 level further down. So, inside `/usr/libexec/qm/` or if it's only to be used in conjunction with podman, maybe in `/usr/libexec/podman/`.

> I am thinking of renaming this repo to just qm

Dunno, I guess having the `-selinux` in the name provides a better idea, but not a stong opinion. We could check with the team if you prefer.

Reply:
> Yes, /usr/libexec/qm/install would be good. Or /usr/lib/qm/install since this is not to be executed by other users of a tool.
> 
> As far as the -selinux drop is concerned, the goal of this repo is setting up a QM isolated environment. The SELinux component is just a small part of it.

SGTM